### DISCUSSION TITLE
How are Visual Studio compatible builds generated?

### DISCUSSION THREAD

**Discussion by mahonbt1** [coin-or/Ipopt#580] [Q&A] [answered]
The prebuilt libraries given are very useful, but the process for generating them ourselves from source isn't in the documentation from what I can tell.  Is it simple?  How do you deal with the fortran dependencies of say MUMPS running under the hood?

Specifically, I need to know how to build my application in an msvc 2017 environment (if possible) by linking against ipopt. The existing process here is to link against the precompiled libraries for our msvc build (via a '-DIPOPT_DIR=' command in our cmake build that points to ipopt.dll.lib ), but if I can take ownership of building for a specific msvc build target and just get it documented either internally on our end or here on github, that could save us some trouble in the future.

**Comment by svigerske** [comment, marked as answer]
Instructions for builds on Windows are in the docu: https://coin-or.github.io/Ipopt/INSTALL.html#SYSTEMPACKAGES_MSYS

The libraries attached to the releases are build this way by using the Intel C/C++/Fortran compilers.
Building Ipopt itself with Microsofts C/C++ compilers would work the same way as with the Intel compilers, but you will need to provide a linear solver still. Without a Fortran compiler, I think the only option on Windows is to use [Pardiso from Intel MKL](https://coin-or.github.io/Ipopt/INSTALL.html#DOWNLOAD_PARDISOMKL).

**Reply by mahonbt1** [reply]
I agree that it's there, but I'm a bit stumped on how to do this exactly.  If we manage to get Thirdparty-MUMPS compiled as a static library via say MinGW's version of gfortran, is it straightforward to build IPOPT with msvc and just point to the static lib during the linking stage?
We have a need to put this onto some embedded targets in the future too so I'm trying to master some of these build details.

**Reply by svigerske** [reply]
Pointing the Ipopt build to your build of MUMPS should be relatively easy (use the `--with-mumps-cflags` and `--with-mumps-lflags` arguments of Ipopt's configure). I cannot say for sure that linking libs build by gfortran and msvc will work. Maybe it does, you'll have to try.
Mixing Intel Fortran (which should be free nowadays) and MSVC usually works fine.

For #584, I'll see whether we can build the MUMPS lib for [the downloads](https://github.com/coin-or/Ipopt/releases) as DLL. You could then pick the MUMPS DLL from there and build Ipopt by yourself (still need to provide some form of Lapack, though).
